src/hooks/useOptimisticGovernorRules.ts

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from "react";
+import useClient from "hooks/useOracleClient";
+import useReader from "hooks/useOracleReader";
+import { ethers } from "ethers";
+
+const ogAbi = [
+  {
+    inputs: [],
+    name: "rules",
+    outputs: [
+      {
+        internalType: "string",
+        name: "",
+        type: "string",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+export function useOptimisticGovernorRules() {
+  const { state } = useClient();
+  const { provider, requester } = useReader(state);
+  const [rules, setRules] = useState<undefined | string>();
+
+  useEffect(() => {
+    if (requester === undefined) return;
```

Comment:
if provider could be undefined we could also make a line to check that and return early

Reply:
Updated to check if provider is undefined.